tests/test_mozillaclub.py

```diff
@@ -0,0 +1,287 @@
+            self.assertEqual(event['uuid'], expected[x][0])
+            self.assertEqual(event['updated_on'], expected[x][1])
+            self.assertEqual(event['category'], 'event')
+            self.assertEqual(event['tag'], 'http://example.com/feed')
+            self.assertEqual(event['data']['Club Name'], expected[x][2])
+
+    @httpretty.activate
+    def test_empty_cells(self):
+        """
+        Test whether the empty fields cells that are not included in the json
+        feed, are filled with None in the generated event.
+        """
+
+        http_requests = configure_http_server()
+
+        # https://docs.google.com/spreadsheets/d/1QHl2bjBhMslyFzR5XXPzMLdzzx7oeSKTbgR5PM8qp64/pubhtml
+        # Club Link is always empty so it is not included in
+        # https://spreadsheets.google.com/feeds/cells/1QHl2bjBhMslyFzR5XXPzMLdzzx7oeSKTbgR5PM8qp64/ohaibtm/public/values?alt=json
+        # Check that the field 'Club Link' exist in the items with None value
+
+        mozillaclub = MozillaClub(MozillaClub_FEED_URL)
+        events = [event for event in mozillaclub.fetch()]
+        self.assertEqual(len(events), 92)
+        for event in events:
+            self.assertEqual('Club Link' in event['data'].keys(), True)
+            self.assertEqual(event['data']['Club Link'], None)
+
+    @httpretty.activate
+    def test_fetch_empty(self):
+        """Test whether it works when no events are fetched"""
+
+        body = """
+            {
+                "encoding": "UTF-8",
+                "feed": {
+                "event": []
+                }
+            }
+        """
+        httpretty.register_uri(httpretty.GET,
+                               MozillaClub_FEED_URL,
+                               body=body, status=200)
+
+        mozillaclub = MozillaClub(MozillaClub_FEED_URL)
+        events = [event for event in mozillaclub.fetch()]
+
+        self.assertEqual(len(events), 0)
+
+
+class TestMozillaClubBackendCache(unittest.TestCase):
+    """MozillaClub backend tests using a cache"""
+
+    def setUp(self):
+        self.tmp_path = tempfile.mkdtemp(prefix='perceval_')
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_path)
+
+    @httpretty.activate
+    def test_fetch_from_cache(self):
+        """Test whether the cache works"""
+
+        http_requests = configure_http_server()
+
+        # First, we fetch the events from the server, storing them
+        # in a cache
+        cache = Cache(self.tmp_path)
+        mozillaclub = MozillaClub(MozillaClub_FEED_URL, cache=cache)
+
+        events = [event for event in mozillaclub.fetch()]
+        self.assertEqual(len(http_requests), 1)
+
+        # Now, we get the events from the cache.
+        # The contents should be the same and there won't be
+        # any new request to the server
+        cached_events = [event for event in mozillaclub.fetch_from_cache()]
+        self.assertEqual(len(cached_events), len(events))
```

Comment:
After reviewing it I compare the full contents in next line with:

    for i in range(0,len(events)):
            self.assertDictEqual(cached_events[i]['data'], events[i]['data'])

Maybe I am missing something ...